src/Storages/ObjectStorage/DataLakes/Iceberg/Utils.cpp

```diff
@@ -84,7 +84,14 @@ std::string getProperFilePathFromMetadataInfo(std::string_view data_path, std::s
     }
     else
     {
-        throw ::DB::Exception(DB::ErrorCodes::BAD_ARGUMENTS, "Expected to find '{}' in data path: '{}'", common_path, data_path);
+        /// Data files can have different path
+        pos = data_path.find("://");
```

Comment:
The method [docs](https://github.com/Altinity/ClickHouse/pull/860/files#diff-7623b16c5b0b217826bbb22df9dbbe7ee6f3f9442223651049074670e0ab6d9fL28) say:

> // For example, if the full blob path is s3://bucket/table_name/data/00000-1-1234567890.avro, the function will return table_name/data/00000-1-1234567890.avro

Which means the `bucket` shall not be included in the return value as far as I understand.

As far as I understand your exception handling,you skip the initial `"://"` and then copy everything after the next `"/"`. But what if you are using [Path-style](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html#path-style-access) URIs (i.e, https://s3.region-code.amazonaws.com/bucket-name/key-name)?

Reply:
Try to remove bucket